test: increase test code coverage in `blas/base/snrm2`

ShabiShett07 opened the pull request:
---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes. report:
  - task: lint_filenames status: passed
  - task: lint_editorconfig status: passed
  - task: lint_markdown status: na
  - task: lint_package_json status: na
  - task: lint_repl_help status: na
  - task: lint_javascript_src status: na
  - task: lint_javascript_cli status: na
  - task: lint_javascript_examples status: na
  - task: lint_javascript_tests status: passed
  - task: lint_javascript_benchmarks status: na
  - task: lint_python status: na
  - task: lint_r status: na
  - task: lint_c_src status: na
  - task: lint_c_examples status: na
  - task: lint_c_benchmarks status: na
  - task: lint_c_tests_fixtures status: na
  - task: lint_shell status: na
  - task: lint_typescript_declarations status: na
  - task: lint_typescript_tests status: na
  - task: lint_license_headers status: passed ---

none

## Description

> What is the purpose of this pull request?

This pull request:

-   Adds test cases for `blas/base/snrm2`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   none

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

ShabiShett07 committed:
test: completes code coverage in blas/base/snrm2

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 committed:
chore: fix test cases

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

kgryte commented:
@ShabiShett07 Are you able to add a couple more tests in order to exercise all branches? Currently, we only have 94% branch coverage.

ShabiShett07 committed:
chore: add test cases

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

ShabiShett07 commented:
@kgryte, I have add test cases, now we have 100% coverage rate

kgryte commented on lib/node_modules/@stdlib/blas/base/snrm2/test/test.ndarray.native.js:
```diff
@@ -110,6 +136,23 @@ tape( 'if provided an `N` parameter less than or equal to `0`, the function retu
 	t.end();
 });
 
+tape( 'the function returns NaN if provided an array element equal to NaN', function test( t ) {
```
```suggestion
tape( 'the function returns NaN if provided an array element equal to NaN', opts, function test( t ) {
```

You need to include the `opts` argument in `*.native.js` tests in order to guard against the scenario where the native add-on has not been built.

kgryte commented on lib/node_modules/@stdlib/blas/base/snrm2/test/test.snrm2.native.js:
```diff
@@ -110,6 +136,23 @@ tape( 'if provided an `N` parameter less than or equal to `0`, the function retu
 	t.end();
 });
 
+tape( 'the function returns NaN if provided an array element equal to NaN', function test( t ) {
```
```suggestion
tape( 'the function returns NaN if provided an array element equal to NaN', opts, function test( t ) {
```

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented on lib/node_modules/@stdlib/blas/base/snrm2/test/test.ndarray.js:
```diff
@@ -83,6 +84,31 @@ tape( 'the function calculates the L2-norm of a vector', function test( t ) {
 	z = snrm2( x.length, x, 1, 0 );
 	isApprox( t, z, 4.0, 1.0 );
 
+	x = new Float32Array( [ 9e-20, 9e-20, 1.2e-19 ] );
```
```suggestion
	x = new Float32Array( [ 9.0e-20, 9.0e-20, 1.2e-19 ] );
```

kgryte commented on lib/node_modules/@stdlib/blas/base/snrm2/test/test.ndarray.native.js:
```diff
@@ -92,6 +93,31 @@ tape( 'the function calculates the L2-norm of a vector', opts, function test( t
 	z = snrm2( x.length, x, 1, 0 );
 	isApprox( t, z, 4.0, 1.0 );
 
+	x = new Float32Array( [ 9e-20, 9e-20, 1.2e-19 ] );
```
```suggestion
	x = new Float32Array( [ 9.0e-20, 9.0e-20, 1.2e-19 ] );
```

kgryte commented on lib/node_modules/@stdlib/blas/base/snrm2/test/test.snrm2.js:
```diff
@@ -83,6 +84,31 @@ tape( 'the function calculates the L2-norm of a vector', function test( t ) {
 	z = snrm2( x.length, x, 1 );
 	isApprox( t, z, 4.0, 1.0 );
 
+	x = new Float32Array( [ 9e-20, 9e-20, 1.2e-19 ] );
```
```suggestion
	x = new Float32Array( [ 9.0e-20, 9.0e-20, 1.2e-19 ] );
```

kgryte commented on lib/node_modules/@stdlib/blas/base/snrm2/test/test.snrm2.native.js:
```diff
@@ -92,6 +93,31 @@ tape( 'the function calculates the L2-norm of a vector', opts, function test( t
 	z = snrm2( x.length, x, 1 );
 	isApprox( t, z, 4.0, 1.0 );
 
+	x = new Float32Array( [ 9e-20, 9e-20, 1.2e-19 ] );
```
```suggestion
	x = new Float32Array( [ 9.0e-20, 9.0e-20, 1.2e-19 ] );
```

kgryte committed:
Apply suggestions from code review

Signed-off-by: Athan <kgryte@gmail.com>

kgryte commented:
@ShabiShett07 Do we need to add similar `NaN` tests to `dnrm2`?
